homeassistant/components/verisure/__init__.py

```diff
@@ -1,116 +1,155 @@
-                vol.Optional(CONF_CODE_DIGITS, default=4): cv.positive_int,
-                vol.Optional(CONF_DOOR_WINDOW, default=True): cv.boolean,
-                vol.Optional(CONF_GIID): cv.string,
-                vol.Optional(CONF_HYDROMETERS, default=True): cv.boolean,
-                vol.Optional(CONF_LOCKS, default=True): cv.boolean,
-                vol.Optional(CONF_DEFAULT_LOCK_CODE): cv.string,
-                vol.Optional(CONF_MOUSE, default=True): cv.boolean,
-                vol.Optional(CONF_SMARTPLUGS, default=True): cv.boolean,
-                vol.Optional(CONF_THERMOMETERS, default=True): cv.boolean,
-                vol.Optional(CONF_SMARTCAM, default=True): cv.boolean,
-                vol.Optional(CONF_SCAN_INTERVAL, default=DEFAULT_SCAN_INTERVAL): (
-                    vol.All(cv.time_period, vol.Clamp(min=MIN_SCAN_INTERVAL))
-                ),
-            }
-        )
-    },
+    vol.All(
+        cv.deprecated(DOMAIN),
+        {
+            DOMAIN: vol.Schema(
+                {
+                    vol.Required(CONF_PASSWORD): cv.string,
+                    vol.Required(CONF_USERNAME): cv.string,
+                    vol.Optional(CONF_CODE_DIGITS): cv.positive_int,
+                    vol.Optional(CONF_GIID): cv.string,
+                    vol.Optional(CONF_DEFAULT_LOCK_CODE): cv.string,
+                },
+                extra=vol.ALLOW_EXTRA,
+            )
+        },
+    ),
     extra=vol.ALLOW_EXTRA,
 )
 
+
 DEVICE_SERIAL_SCHEMA = vol.Schema({vol.Required(ATTR_DEVICE_SERIAL): cv.string})
 
 
-async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
+async def async_setup(hass: HomeAssistant, config: dict[str, Any]) -> bool:
     """Set up the Verisure integration."""
-    verisure = Verisure(config[DOMAIN][CONF_USERNAME], config[DOMAIN][CONF_PASSWORD])
-    coordinator = VerisureDataUpdateCoordinator(
-        hass, session=verisure, domain_config=config[DOMAIN]
-    )
+    if DOMAIN in config:
+        hass.async_create_task(
+            hass.config_entries.flow.async_init(
+                DOMAIN,
+                context={"source": SOURCE_IMPORT},
+                data={
+                    CONF_EMAIL: config[DOMAIN][CONF_USERNAME],
+                    CONF_PASSWORD: config[DOMAIN][CONF_PASSWORD],
+                    CONF_GIID: config[DOMAIN].get(CONF_GIID),
+                    CONF_LOCK_CODE_DIGITS: config[DOMAIN].get(CONF_CODE_DIGITS),
+                    CONF_LOCK_DEFAULT_CODE: config[DOMAIN].get(CONF_LOCK_DEFAULT_CODE),
+                },
+            )
+        )
+
+    return True
 
-    if not await hass.async_add_executor_job(coordinator.login):
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up Verisure from a config entry."""
+    # Migrate old YAML settings (hidden in the config entry),
+    # to config entry options. Can be removed after YAML support is gone.
+    if (
```

Comment:
Keep a top level if to see if options is empty to contain all legacy logic?

Reply:
That actually doesn't work, as even after all this you might still end up with empty options 😉 

This is the same logic as used initially by Hue, however, in this case it can be removed after two releases.

Reply:
Did change the logic now a bit; keeping all migration stuff in a single if branch, reduces the entry updates and ensures the data entry is always cleaned up.